Core/src/Propagator/detail/CovarianceEngine.cpp

```diff
@@ -15,6 +15,53 @@
 
 namespace Acts {
 namespace {
+/// @brief Perform a tiled matrix multiplication, avoiding Eigen GEMM methods.
+///
+/// @tparam M First dimension of first matrix
+/// @tparam N Second dimension of first matrix, and first dimension of second
+/// matrix
+/// @tparam P Second dimension of second matrix
+///
+/// @param[in] a MxN matrix
+/// @param[in] b NxP matrix
+///
+/// @returns The product ab
+template <std::size_t M, std::size_t N, std::size_t P>
```

Comment:
I guess these could be deduced from the matrix arguments?

Reply:
I thought so too, but surprisingly it struggled a little. However, we should be able to accept the matrix types themselves (not their sizes) as the template types, and _then_ it should be automatically deducible.

Reply:
Yeah right. That coupled with a few `static_assert`s and it should be good. Up to you, could also do this in a follow-up PR.